pkgs/development/compilers/dotnet/combinePackages.nix

```diff
@@ -0,0 +1,20 @@
+{ packages }: 
+{ buildEnv, lib }:
+let cli = builtins.head packages;
+in
+assert lib.assertMsg ((builtins.length packages) != 0) 
+    ''You must include at least one package, e.g 
+      `with dotnetCorePackages; combinePackages {
+          packages = [ sdk_3_0 aspnetcore_2_1 ];
+       };`'' ;
```

Comment:
You'll probably want to switch the order of the assert and let. Otherwise you'll get the head error from having an empty list rather than the helpful message:

```
assert lib.assertMsg ((builtins.length packages) != 0) 
  ''
    ...
  '';
let
  cli = builtins.head packages;
in
```